nemo/collections/nlp/modules/common/megatron/transformer.py

```diff
@@ -620,6 +621,17 @@ def forward(
         real_seq_length = hidden_states.shape[0]
         key_length = key_layer.shape[0]
 
+        if prefix_tuning_key_values is not None:
+            key, value = prefix_tuning_key_values[0], prefix_tuning_key_values[1]
+            key_layer = torch.cat((key, key_layer), dim=0)
+            value_layer = torch.cat((value, value_layer), dim=0)
+            extra_attention_mask = torch.zeros(
+                (attention_mask.shape[0], 1, attention_mask.shape[-2], key.shape[0]),
+                dtype=torch.bool,
+                device=attention_mask.device,
+            )
+            attention_mask = torch.cat((extra_attention_mask, attention_mask), dim=-1)
+
```

Comment:
This is ok for now, but ideally we can reuse `layer_past` to do this. For `extra_attn_mask` we can create an appropriate `attn_mask` during dataloader stage. @yidong72